### ISSUE TITLE
[BUG] [Upgrade] Elasticsearch settings hard-coded instead of read from existing configuration file

### ISSUE THREAD

**Issue by to-bar** [epiphany-platform/epiphany#2244]
**Describe the bug**
In upgrade role the following Elasticsearch settings are hard-coded:
- http.port: 9200
- transport.port: 9300
- network.host: "{{ ansible_default_ipv4.address | default(ansible_all_ipv4_addresses[0]) }}"

If any of them has custom value in config file, then `epicli upgrade` will fail.

**How to reproduce**
1. Deploy a cluster using old `epicli` version (e.g. v0.8.2).
2. Edit /etc/elasticsearch/elasticsearch.yml setting a custom value for any property from the mentioned above.
3. Run `systemctl restart elasticsearch.service`
4. Try to upgrade Elasticsearch using `epicli` v0.10.0.

**Expected behavior**
Mentioned settings should be read from existing config file.

**Environment**
- Cloud provider: All
- OS: All

**epicli version**: 0.10.0

**Additional context**
n/a

---

**DoD checklist**

* [x] Changelog updated (if affected version was released)
* [x] COMPONENTS.md updated / doesn't need to be updated
* [x] Automated tests passed (QA pipelines)
  * [x] apply
  * [x] upgrade
* [ ] Case covered by automated test (if possible)
* [ ] Idempotency tested
* [x] Documentation updated / doesn't need to be updated
* [x] All conversations in PR resolved
* [x] Backport tasks created / doesn't need to be backported

**Comment by przemyslavic** [comment]
:heavy_check_mark: It works fine when using default (hard-coded) ports: tested `apply` and `upgrade` form v0.8 to develop.

However there is a problem upgrading elasticsearch when following the above scenario:
1. Deployed a cluster using epicli v0.8.
2. Edited /etc/elasticsearch/elasticsearch.yml setting a custom value for:
`http.port: 9201`
`transport.port: 9301`
for 2 nodes
3. Added rules for ports 9201 and 9301 to security group
4. Ran `systemctl restart elasticsearch.service`
5. Tried to upgrade Elasticsearch to latest develop

:x: It failed:
```
09:50:13 INFO cli.engine.ansible.AnsibleCommand - PLAY [logging] *****************************************************************
09:50:13 INFO cli.engine.ansible.AnsibleCommand -
09:50:13 INFO cli.engine.ansible.AnsibleCommand - TASK [Gathering Facts] *********************************************************
09:50:15 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:15 INFO cli.engine.ansible.AnsibleCommand -
09:50:15 INFO cli.engine.ansible.AnsibleCommand - TASK [preflight_facts : Load preflight facts] **********************************
09:50:15 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:15 INFO cli.engine.ansible.AnsibleCommand -
09:50:15 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Open Distro for Elasticsearch | Get information about installed packages as facts] ***
09:50:17 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:17 INFO cli.engine.ansible.AnsibleCommand -
09:50:17 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Open Distro for Elasticsearch | Assert that elasticsearch-oss package is installed] ***
09:50:17 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:17 INFO cli.engine.ansible.AnsibleCommand -
09:50:17 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Open Distro for Elasticsearch | Include defaults from opendistro_for_elasticsearch role] ***
09:50:17 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:17 INFO cli.engine.ansible.AnsibleCommand -
09:50:17 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Open Distro for Elasticsearch | Print elasticsearch-oss versions] ***
09:50:17 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com] => {
09:50:17 INFO cli.engine.ansible.AnsibleCommand -     "msg": [
09:50:17 INFO cli.engine.ansible.AnsibleCommand -         "Installed version: 7.9.1",
09:50:17 INFO cli.engine.ansible.AnsibleCommand -         "Target version: 7.10.2"
09:50:17 INFO cli.engine.ansible.AnsibleCommand -     ]
09:50:17 INFO cli.engine.ansible.AnsibleCommand - }
09:50:17 INFO cli.engine.ansible.AnsibleCommand -
09:50:17 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Open Distro for Elasticsearch | Check if upgrade state file exists] ***
09:50:18 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:19 INFO cli.engine.ansible.AnsibleCommand -
09:50:19 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Open Distro for Elasticsearch | Upgrade Elasticsearch and ODFE plugins] ***
09:50:19 INFO cli.engine.ansible.AnsibleCommand - included: /shared/build/testopendistro2/ansible/roles/upgrade/tasks/opendistro_for_elasticsearch/upgrade-elasticsearch.yml for ec2-1-1-1-1.eu-west-1.compute.amazonaws.com
09:50:19 INFO cli.engine.ansible.AnsibleCommand -
09:50:19 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Create upgrade state file] ******************************
09:50:20 INFO cli.engine.ansible.AnsibleCommand - changed: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:20 INFO cli.engine.ansible.AnsibleCommand -
09:50:20 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Ensure elasticsearch service is running] ****************
09:50:22 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:22 INFO cli.engine.ansible.AnsibleCommand -
09:50:22 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : include_tasks] *************************************************
09:50:22 INFO cli.engine.ansible.AnsibleCommand - included: /shared/build/testopendistro2/ansible/roles/upgrade/tasks/opendistro_for_elasticsearch/utils/get-config-from-files.yml for ec2-1-1-1-1.eu-west-1.compute.amazonaws.com
09:50:22 INFO cli.engine.ansible.AnsibleCommand -
09:50:22 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Load existing elasticsearch.yml] ************************
09:50:23 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:23 INFO cli.engine.ansible.AnsibleCommand -
09:50:23 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Get Xmx value from /etc/elasticsearch/jvm.options] ******
09:50:25 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:25 INFO cli.engine.ansible.AnsibleCommand -
09:50:25 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Set existing configuration facts] ***********************
09:50:25 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:25 INFO cli.engine.ansible.AnsibleCommand -
09:50:25 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Set common facts] ***************************************
09:50:25 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:25 INFO cli.engine.ansible.AnsibleCommand -
09:50:25 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Wait for elasticsearch service to start up] *************
09:50:25 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:25 INFO cli.engine.ansible.AnsibleCommand -
09:50:25 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Gather facts on listening ports] *******************************
09:50:25 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:25 INFO cli.engine.ansible.AnsibleCommand -
09:50:25 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Get host address based on transport port] **********************
09:50:25 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:25 INFO cli.engine.ansible.AnsibleCommand -
09:50:25 INFO cli.engine.ansible.AnsibleCommand - TASK [include_role : certificate] **********************************************
09:50:25 INFO cli.engine.ansible.AnsibleCommand -
09:50:25 INFO cli.engine.ansible.AnsibleCommand - TASK [certificate : Install requirements for Ansible certificate modules] ******
09:50:29 INFO cli.engine.ansible.AnsibleCommand - changed: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:29 INFO cli.engine.ansible.AnsibleCommand -
09:50:29 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Get information on root-ca certificate] ************************
09:50:30 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:30 INFO cli.engine.ansible.AnsibleCommand -
09:50:30 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Check if demo or epiphany certificates are in use] ******
09:50:30 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:30 INFO cli.engine.ansible.AnsibleCommand -
09:50:30 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Set API endpoint] ***************************************
09:50:30 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:31 INFO cli.engine.ansible.AnsibleCommand -
09:50:31 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Set API access facts] ***********************************
09:50:31 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:31 INFO cli.engine.ansible.AnsibleCommand -
09:50:31 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Test API access using custom certificate] ***************
09:50:31 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:31 INFO cli.engine.ansible.AnsibleCommand -
09:50:31 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Test API access using demo certificate] *****************
09:50:33 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:33 INFO cli.engine.ansible.AnsibleCommand -
09:50:33 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : Fail with informative message] *********************************
09:50:33 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:33 INFO cli.engine.ansible.AnsibleCommand -
09:50:33 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Disable shard allocation for the cluster] ***************
09:50:34 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:34 INFO cli.engine.ansible.AnsibleCommand -
09:50:34 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Perform a synced flush] *********************************
09:50:36 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:36 INFO cli.engine.ansible.AnsibleCommand -
09:50:36 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Stop elasticsearch service] *****************************
09:50:37 INFO cli.engine.ansible.AnsibleCommand - changed: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:37 INFO cli.engine.ansible.AnsibleCommand -
09:50:37 INFO cli.engine.ansible.AnsibleCommand - TASK [ODFE | Include Elasticsearch installation tasks] *************************
09:50:38 INFO cli.engine.ansible.AnsibleCommand -
09:50:38 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Install elasticsearch-oss packages] *******
09:50:53 INFO cli.engine.ansible.AnsibleCommand - changed: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:53 INFO cli.engine.ansible.AnsibleCommand -
09:50:53 INFO cli.engine.ansible.AnsibleCommand - TASK [ODFE | Include Elasticsearch configuration tasks] ************************
09:50:53 INFO cli.engine.ansible.AnsibleCommand -
09:50:53 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Ensure snapshot folder exists] ************
09:50:54 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:54 INFO cli.engine.ansible.AnsibleCommand -
09:50:54 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Provide JVM configuration file] ***********
09:50:56 INFO cli.engine.ansible.AnsibleCommand - changed: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:56 INFO cli.engine.ansible.AnsibleCommand -
09:50:56 INFO cli.engine.ansible.AnsibleCommand - TASK [include_role : certificate] **********************************************
09:50:56 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:57 INFO cli.engine.ansible.AnsibleCommand -
09:50:57 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : include_tasks] ****************************
09:50:57 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:50:57 INFO cli.engine.ansible.AnsibleCommand -
09:50:57 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Provide Elasticsearch configuration file] ***
09:51:00 INFO cli.engine.ansible.AnsibleCommand - changed: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:00 INFO cli.engine.ansible.AnsibleCommand -
09:51:00 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Restart elasticsearch service] ************
09:51:17 INFO cli.engine.ansible.AnsibleCommand - changed: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:17 INFO cli.engine.ansible.AnsibleCommand -
09:51:17 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Enable and start elasticsearch service] ***
09:51:18 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:18 INFO cli.engine.ansible.AnsibleCommand -
09:51:18 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Wait for elasticsearch service to start up] ***
09:51:18 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:18 INFO cli.engine.ansible.AnsibleCommand -
09:51:18 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Set helper facts] *************************
09:51:18 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:19 INFO cli.engine.ansible.AnsibleCommand -
09:51:19 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Check if default admin user exists] *******
09:51:19 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:19 INFO cli.engine.ansible.AnsibleCommand -
09:51:19 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Set OpenDistro admin password] ************
09:51:19 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:19 INFO cli.engine.ansible.AnsibleCommand -
09:51:19 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Check if default kibanaserver user exists] ***
09:51:19 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:19 INFO cli.engine.ansible.AnsibleCommand -
09:51:19 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Set OpenDistro kibanaserver password] *****
09:51:19 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:19 INFO cli.engine.ansible.AnsibleCommand -
09:51:19 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Check if default logstash user exists] ****
09:51:19 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:19 INFO cli.engine.ansible.AnsibleCommand -
09:51:19 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Set OpenDistro logstash password] *********
09:51:19 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:19 INFO cli.engine.ansible.AnsibleCommand -
09:51:19 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Remove OpenDistro demo users] *************
09:51:19 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:19 INFO cli.engine.ansible.AnsibleCommand -
09:51:19 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Include upgrade plugins tasks] **************************
09:51:19 INFO cli.engine.ansible.AnsibleCommand - included: /shared/build/testopendistro2/ansible/roles/upgrade/tasks/opendistro_for_elasticsearch/upgrade-plugins.yml for ec2-1-1-1-1.eu-west-1.compute.amazonaws.com
09:51:19 INFO cli.engine.ansible.AnsibleCommand -
09:51:19 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE plugins | Assert that opendistro-* packages are installed] ***
09:51:19 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com] => (item=opendistro-alerting)
09:51:19 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com] => (item=opendistro-index-management)
09:51:19 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com] => (item=opendistro-job-scheduler)
09:51:20 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com] => (item=opendistro-performance-analyzer)
09:51:20 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com] => (item=opendistro-security)
09:51:20 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com] => (item=opendistro-sql)
09:51:20 INFO cli.engine.ansible.AnsibleCommand -
09:51:20 INFO cli.engine.ansible.AnsibleCommand - TASK [ODFE plugins | Upgrade opendistro-* packages] ****************************
09:51:20 INFO cli.engine.ansible.AnsibleCommand -
09:51:20 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Install opendistro-* packages] ************
09:51:21 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:21 INFO cli.engine.ansible.AnsibleCommand -
09:51:21 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Restart elasticsearch service] **************************
09:51:21 INFO cli.engine.ansible.AnsibleCommand - skipping: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]
09:51:21 INFO cli.engine.ansible.AnsibleCommand -
09:51:21 INFO cli.engine.ansible.AnsibleCommand - TASK [upgrade : ODFE | Wait for elasticsearch node to come back up] ************

09:56:23 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]: FAILED! => {"changed": false, "elapsed": 300, "msg": "Timeout when waiting for ec2-1-1-1-1:9301"}
09:56:23 INFO cli.engine.ansible.AnsibleCommand -
09:56:23 INFO cli.engine.ansible.AnsibleCommand - PLAY RECAP *********************************************************************
09:56:23 INFO cli.engine.ansible.AnsibleCommand - ec2-1-1-1-1.eu-west-1.compute.amazonaws.com : ok=36   changed=7    unreachable=0    failed=1    skipped=47   rescued=0    ignored=0
09:56:23 INFO cli.engine.ansible.AnsibleCommand - ec2-3-3-3-3.eu-west-1.compute.amazonaws.com : ok=4    changed=0    unreachable=0    failed=0    skipped=25   rescued=0    ignored=0
09:56:23 INFO cli.engine.ansible.AnsibleCommand - ec2-2-2-2-2.eu-west-1.compute.amazonaws.com : ok=4    changed=0    unreachable=0    failed=0    skipped=25   rescued=0    ignored=0
09:56:23 INFO cli.engine.ansible.AnsibleCommand -
09:56:23 INFO cli.engine.ansible.AnsibleRunner - Run done in 5627768ms
09:56:23 INFO cli.engine.UpgradeEngine - Run done in 5655479ms
```
It looks like it's still using default ports 9200 and 9300 during the upgrade and does not correctly read/apply the configuration from the remote host.
```
# ---------------------------------- Network -----------------------------------
#
# Set the bind address to a specific IP (IPv4 or IPv6):
#
network.host: ec2-18-200-196-228
#
# Set a custom port for HTTP:
#
http.port: 9200
#
transport.port: 9300
#
# For more information, consult the network module documentation.
#
```
```
root@ec2-1-1-1-1:~# ss -tunl
Netid           State             Recv-Q            Send-Q                             Local Address:Port                       Peer Address:Port
udp             UNCONN            0                 0                                  127.0.0.53%lo:53                              0.0.0.0:*
udp             UNCONN            0                 0                                 10.1.3.14%ens5:68                              0.0.0.0:*
tcp             LISTEN            0                 128                                127.0.0.53%lo:53                              0.0.0.0:*
tcp             LISTEN            0                 128                                      0.0.0.0:22                              0.0.0.0:*
tcp             LISTEN            0                 128                                    10.1.3.14:5601                            0.0.0.0:*
tcp             LISTEN            0                 128                                         [::]:22                                 [::]:*
tcp             LISTEN            0                 128                                            *:9100                                  *:*
tcp             LISTEN            0                 128                           [::ffff:10.1.3.14]:9200                                  *:*
tcp             LISTEN            0                 128                           [::ffff:10.1.3.14]:9300                                  *:*
```

Moving back to TODO.

**Comment by to-bar** [comment]
Upgrade role has been refactored, and this bug may be fixed by #2325

**Comment by to-bar** [comment]
Fixed in PR #2341

**Comment by przemyslavic** [comment]
✔️ Successfully upgraded `elasticsearch` by following the scenario described above.
- `epicli upgrade` from v0.8 to develop
